### DISCUSSION TITLE
New client cannot be used to manage the master realm. Can I enable a non-built-in client for master realm management?

### DISCUSSION THREAD

**Discussion by moritzschmitz-oviva** [keycloak/keycloak#26896] [Q&A] [answered]
Started with a fresh new installation of Keycloak.

I created a new client `keycloak-config-cli` in the master realm. Also a user `keycloak-config-cli` with the `admin` role.

Now I use the tool with the same name to import my realm config via the created client. I keep getting a `403` on the `/admin/serverinfo` path though.

Can I even use a different client than the `admin-cli` to manage the master realm? Do I need to enable some setting for this, because I couldn't find any?

Client config (copied from `admin-cli` client):
```
{
  "clientId": "keycloak-config-cli",
  "standardFlowEnabled": false,
  "directAccessGrantsEnabled": true,
  "publicClient": true,
  "authenticationFlowBindingOverrides": {
    "direct_grant": "direct grant x509"
  },
  "fullScopeAllowed": false,
  "nodeReRegistrationTimeout": 0
}
```

**Comment by danielFesenmeyer** [comment, marked as answer]
The clients admin-cli and security-admin-console are handled in a special way on the server: Permissions are read directly from the database - NOT from the token, as for all the other clients.
If you don't have that many permissions per user (not so many that the token gets too large), you can set fullScopeAllowed to true on the client - otherwise you have to select which roles should be included (probably some of the roles of the master realm-management client).

**Reply by moritzschmitz-oviva** [reply]
Works now. Thanks for the explanation!
